core/rawdb/freezer_table.go

```diff
@@ -483,6 +485,12 @@ func (t *freezerTable) Retrieve(item uint64) ([]byte, error) {
 	return snappy.Decode(nil, blob)
 }
 
+// has returns an indicator whether the specified number data
+// exists in the freezer table.
+func (t *freezerTable) has(number uint64) bool {
+	return atomic.LoadUint64(&t.items) > number
```

Comment:
```suggestion
	return atomic.LoadUint64(&t.items) >= number
```
I'm not 100% certain, but 99

Reply:
nevermind

Reply:
`items      uint64        // Number of items stored in the table`

In append:

```go
	// Ensure only the next item can be written, nothing else
	if atomic.LoadUint64(&t.items) != item {
		panic(fmt.Sprintf("appending unexpected item: want %d, have %d", t.items, item))
	}
```

i.e. if we insert item `#n`, we set the counter to `n+1`. So the existing check is ok.